src/algebra/group/opposite.lean

```diff
@@ -42,30 +42,38 @@ unop_injective.add_zero_class _ rfl (λ x y, rfl)
 instance [add_monoid α] : add_monoid αᵐᵒᵖ :=
 unop_injective.add_monoid _ rfl (λ _ _, rfl) (λ _ _, rfl)
 
+instance [add_comm_monoid α] : add_comm_monoid αᵐᵒᵖ :=
+unop_injective.add_comm_monoid _ rfl (λ _ _, rfl) (λ _ _, rfl)
+
+@[to_additive] instance [has_nat_cast α] : has_nat_cast αᵐᵒᵖ := ⟨λ n, op n⟩
+@[to_additive] instance [has_int_cast α] : has_int_cast αᵐᵒᵖ := ⟨λ n, op n⟩
```

Comment:
Do you think these belong in `algebra/opposites` instead?

Reply:
`has_nat_cast` doesn't exist there.

Reply:
Ah, ok. Can we put these right at the top of the file instead then? The current design seems to be to get all the data out of the way first (in the other file), and build the algebra instances later.